src/sqlfluff/utils/reflow/sequence.py

```diff
@@ -0,0 +1,450 @@
+        """
+        # There's probably a more efficient way than immediately
+        # materialising the raw_segments for the whole root, but
+        # it works. Optimise later.
+        all_raws = root_segment.raw_segments
+
+        target_raws = target_segment.raw_segments
+        assert target_raws
+        pre_idx = all_raws.index(target_raws[0])
+        post_idx = all_raws.index(target_raws[-1]) + 1
+        initial_idx = (pre_idx, post_idx)
+        if sides in ("both", "before"):
+            # Catch at least the previous segment
+            pre_idx -= 1
+            while pre_idx - 1 > 0 and all_raws[pre_idx].is_type(
+                "whitespace", "newline", "indent"
+            ):
+                pre_idx -= 1
+        if sides in ("both", "after"):
+            while post_idx < len(all_raws) and all_raws[post_idx].is_type(
+                "whitespace", "newline", "indent"
+            ):
+                post_idx += 1
+            # Capture one more after the whitespace.
+            post_idx += 1
+        segments = all_raws[pre_idx:post_idx]
+        reflow_logger.debug(
+            "Generating ReflowSequence.from_around_target(). idx: %s. "
+            "slice: %s:%s. segments: %s",
+            initial_idx,
+            pre_idx,
+            post_idx,
+            segments,
+        )
+        return cls.from_raw_segments(segments, root_segment, config=config)
+
+    def _find_element_idx_with(self, target: RawSegment) -> int:
+        for idx, elem in enumerate(self.elements):
+            if target in elem.segments:
+                return idx
+        raise ValueError(  # pragma: no cover
+            f"Target [{target}] not found in ReflowSequence."
+        )
+
+    def without(self, target: RawSegment) -> "ReflowSequence":
+        """Returns a new reflow sequence without the specified segment.
+
+        It's important to note that this doesn't itself remove the target
+        from the file. This just allows us to simulate a sequence without it
+        and work out what additional whitespace changes would be required
+        if we were to remove it.
+        """
+        removal_idx = self._find_element_idx_with(target)
+        if removal_idx == 0 or removal_idx == len(self.elements) - 1:
+            raise NotImplementedError(  # pragma: no cover
+                "Unexpected removal at one end of a ReflowSequence."
+            )
+        if isinstance(self.elements[removal_idx], ReflowPoint):
+            raise NotImplementedError(  # pragma: no cover
+                "Not expected removal of whitespace in ReflowSequence."
+            )
+        merged_point = ReflowPoint(
+            segments=list(self.elements[removal_idx - 1].segments)
+            + list(self.elements[removal_idx + 1].segments),
+        )
+        return ReflowSequence(
+            elements=list(self.elements[: removal_idx - 1])
+            + [merged_point]
+            + list(self.elements[removal_idx + 2 :]),
```

Comment:
Nitpick: Are the two calls to `list()` call necessary here? Slicing a list already creates a new list.

https://discuss.python.org/t/does-the-list-slicing-produce-a-new-list/14627

Reply:
I did it this way because they're necessary for mypy type checking, but I think you're right - will find a different route.